### ISSUE TITLE
Wrong output shape of torch.nn.Conv2d with 2d stride overload

### ISSUE THREAD

**Issue by 241721** [dotnet/TorchSharp#496]
Hi,
according to python's torch documentation (image below),
with W_in: 8, kernelSize: (1, 7), stride: (1, 1), padding: (0, 3) and default dilation (1,1) i should have recived W_out = 8.
I got 2 instead. That could be an issue I think.
![image](https://user-images.githubusercontent.com/39262993/150590662-fb773287-5885-48d5-aef9-474cb8d7553a.png)

There isn't that problem with H_out, when i set
W_in: 8, kernelSize: (7, 1), stride: (1, 1), padding: (3, 0)
Screen from
https://pytorch.org/docs/1.9.1/generated/torch.nn.Conv2d.html

**Comment by NiklasGustafsson** [comment]
I'm seeing the same thing. Thanks for the issue report.

**Comment by 241721** [comment]
![image](https://user-images.githubusercontent.com/39262993/150862678-e9f5beeb-10b4-424f-961d-e3e6eed81b93.png)
here is the problem
i want to set padding (0,3) but it sets that enum instead

**Comment by NiklasGustafsson** [comment]
Yes, it's the encoding of padding='same' and padding='valid' that is incorrectly received in the C++ code. Same for Conv3d.

**Comment by NiklasGustafsson** [comment]
@241721 -- since you already looked into the details yourself, maybe you could do a quick review of PR #497?

**Comment by NiklasGustafsson** [comment]
@241721 -- thanks for the PR review!

**Comment by 241721** [comment]
Thx for your response, i approved.
